IniFile.cs

```diff
@@ -49,12 +49,23 @@ public static void ConsolidateIniFiles(IniFile firstIni, IniFile secondIni)
         /// <param name="filePath">The path of the INI file.</param>
         public IniFile(string filePath)
         {
-            FileName = filePath;
+            this.FileName = filePath;
 
-            if (File.Exists(filePath))
-            {
-                ParseIniFile(File.OpenRead(filePath));
-            }
+            Parse();
+        }
+
+        /// <summary>
+        /// Creates a new INI file instance and parses it.
+        /// </summary>
+        /// <param name="filePath">The path of the INI file.</param>
+        /// <param name="encoding">The encoding of the INI file. Default for UTF-8.</param>
+        public IniFile(string filePath, System.Text.Encoding encoding)
+        {
+            this.FileName = filePath;
+
+            this.Encoding = encoding;
```

Comment:
No need to include "this" when there's no name conflict with a local variable.

You could've just added `encoding` as an optional argument to the already existing constructor.

Reply:
It is better to avoid optional arguments.
Otherwise, the DLL file must be compiled together with whatever program includes it.
See: CA1026 DefaultParametersShouldNotBeUsed

https://docs.microsoft.com/zh-cn/visualstudio/code-quality/ca1026-default-parameters-should-not-be-used

Reply:
Seems that using default paremeters in DLLs can indeed cause issues. That's news to me, but good to know.